sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/policy/MockPartialResponsePolicy.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.test.shared.policy;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MockPartialResponsePolicy implements HttpPipelinePolicy {
+    static final HttpHeaderName RANGE_HEADER = HttpHeaderName.fromString("x-ms-range");
+    private int tries;
+    private final List<int[]> rangesList = new ArrayList<>();
+
+    public MockPartialResponsePolicy(int tries) {
+        this.tries = tries;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return next.process().flatMap(response -> {
+            HttpHeader rangeHttpHeader = response.getRequest().getHeaders().get(RANGE_HEADER);
```

Comment:
Are we always guaranteed for consumers of this to use `x-ms-range` as the range header? If not, in the future when needed this could be modified to include a header in the constructor to determine which header to look for.